squash/dashboard/serializers.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework import serializers
+from rest_framework.reverse import reverse
+from .models import Job, Metric, Measurement
+from django.db import transaction
+
+
+class MetricSerializer(serializers.ModelSerializer):
+
+    links = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Metric
+        fields = ('metric', 'description', 'units', 'condition',
+                  'minimum', 'design', 'stretch', 'user', 'links',)
+
+    def get_links(self, obj):
+        request = self.context['request']
+        return {
+            'self': reverse('metric-detail', kwargs={'pk': obj.pk},
+                            request=request),
+         }
+
+
+class MeasurementSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Measurement
+        fields = ('metric', 'value',)
+
+    def get_links(self, obj):
+        request = self.context['request']
+        return {
+            'self': reverse('measurement-detail', kwargs={'pk': obj.pk},
+                            request=request),
+        }
+
+
+class JobSerializer(serializers.ModelSerializer):
+
+    links = serializers.SerializerMethodField()
+
+    measurements = MeasurementSerializer(many=True)
+
+    class Meta:
+        model = Job
+        fields = ('name', 'build', 'runtime',
+                  'url', 'status', 'measurements', 'links',)
+
+    """ Override the create method to create nested objects
+    from request data
+    """
+
+    def create(self, data):
+        measurements = data.pop('measurements')
```

Comment:
Yeah sorry about that. It was out of context. This was specifically to address @afausti problem in [this JIRA issue comment](https://jira.lsstcorp.org/browse/DM-5745?focusedCommentId=47227&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-47227).